chore(nix): add flake

cmdoret opened the pull request.

cmdoret committed:
chore(nix): add flake

supermaxiste commented:
Some comments:

1. If we add a Nix flake, I would add some explanation on the README how to use it. Can we temporarily add it to the make commands? Maybe like `make nix-dev`?

2. The current flake is only Linux compatible. When running it on MacOS I get the message below. Let's write that Nix is only for Linux for now.

3. CI: can we add an action that tries to build the Nix env when we change the flake? 

```shell
error: flake 'git+file:///Users/stefan/Documents/kg-llm-interface?dir=tools/nix' does not provide attribute 'packages.aarch64-darwin.default' or 'defaultPackage.aarch64-darwin'
```

cmdoret committed:
docs(nix): instructions to use nix with just

cmdoret committed:
ci(nix): dev shell action

cmdoret commented:
1. Added explanation on Readme + recipe in _justfile_ I figured I might do it in `just` directly, as we're migrating anyways
2. Noted it's linux only for now (does it not work on aarch64 if you `rm` the lock file (tools/nix/flake.lock) and rebuild?
3. added ci for building the devshell whenever a `*.nix` file is touched by a PR

supermaxiste commented:
I shall test the lock removal approach for another time. All looks good, excellent job 🚀
